lib/mix/tasks/group_uploader.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Mix.Tasks.GroupUploader do
+  use Mix.Task
+
+  alias Cadet.{Accounts, Course}
+
+  @moduledoc """
+  Mix Task to upload the tutorial groups and it's
+  participants
+  """
+
+  def run([groups_arg | [avengers_arg | _]]) do
+    Mix.Task.run("app.start")
+    # Removing unneccesary headers
+    groups =
+      groups_arg
+      |> Xlsxir.multi_extract()
+      |> Keyword.get(:ok)
+      |> Xlsxir.get_list()
+      |> Enum.drop(3)
+
+    avengers =
+      avengers_arg
+      |> Xlsxir.multi_extract()
+      |> Keyword.get(:ok)
+      |> Xlsxir.get_list()
+      |> Enum.drop(1)
+
+    helper(groups, 0, avengers, 0, nil)
+  end
+
+  defp helper(rows, row_index, avengers, avenger_index, current_group) do
+    row = Enum.at(rows, row_index)
+    current_avenger = Enum.at(avengers, avenger_index)
+
+    cond do
```

Comment:
Please use pattern-matching for this. Quite unreadable tbh. Will review the logic after refactoring.